### ISSUE TITLE
Adaptive gamma "get_gamma" is nowhere used

### ISSUE THREAD

**Issue by Jordy-VL** [torrvision/focal_calibration#13]
https://github.com/torrvision/focal_calibration/blob/475422a6754e20d25216158a874749c33e826e21/Losses/focal_loss_adaptive_gamma.py#L16

Hi guys, 

Thanks for the great repo. I implemented your adaptive focal loss function in tensorflow, yet I noticed that the "get_gamma" function is nowhere used. Rather the get_gamma_list uses fixed gamma values for fixed probabilities, the so-called F532 schedule. 

If I would want to switch to this adaptive gamma, how should I go about this? 

My current code: 
```
ps = [0.2, 0.5]
gammas = [5.0, 3.0]
i = 0
gamma_dic = {}
for p in ps:
    gamma_dic[p] = gammas[i]
    i += 1

def get_gamma_list(pt, default_gamma=2):
    gamma_tensor = tf.ones_like(tf.shape(pt)) * default_gamma
    for key in sorted(gamma_dic.keys(), reverse=True):
        gamma_tensor = tf.where(tf.less_equal(pt, tf.cast(key,tf.float32)), tf.cast(gamma_dic[key], tf.int32), gamma_tensor)
    return tf.cast(gamma_tensor, dtype=tf.float32)
```

Essentially, for each argmax probability in pt (batchsize), I should apply this function? 

Thanks in advance!

**Comment by vivkul** [comment]
Hi Jordy,

Thanks for going over the code!
That's right, it seems correct way to implement F532 if default_gamma=2 and F53 if default_gamma=3.
We used `get_gamma` function just to get the gammas corresponding to ps = [0.2, 0.5]. Once we had these gamma values, we didn't need this function anymore to avoid redundant computation. I hope it makes clear.

Thanks!
